frontend/src/metabase/query_builder/components/view/View/NotebookContainer/NotebookContainer.tsx

```diff
@@ -37,32 +47,89 @@ export const NotebookContainer = ({
     }
   };
 
+  const dispatch = useDispatch();
+  const handleResizeStop = (
+    _event: SyntheticEvent,
+    data: ResizeCallbackData,
+  ) => {
+    dispatch(setUIControls({ nativePreviewSidebarWidth: data.size.width }));
+  };
+
   const transformStyle = isOpen ? "translateY(0)" : "translateY(-100%)";
 
+  const Handle = forwardRef<HTMLDivElement, Partial<ResizableBoxProps>>(
+    function Handle(props, ref) {
+      const handleWidth = 6;
+      // 0.5 accounts for the border width of 1px
+      const left = rem(-(handleWidth / 2 + 0.5));
+
+      return (
+        <Box
+          ref={ref}
+          {...props}
+          pos="absolute"
+          top={0}
+          bottom={0}
+          m="auto 0"
+          h={rem(100)}
+          w={rem(handleWidth)}
+          left={left}
+          bg={darken("border", 0.03)}
+          style={{
+            zIndex: 5,
+            cursor: "ew-resize",
+            borderRadius: rem(8),
+          }}
+        ></Box>
+      );
+    },
+  );
+
   return (
     <Flex
-      className={NC.notebookContainer}
-      bg="white"
+      pos="absolute"
+      inset={0}
+      bg={color("white")}
       opacity={isOpen ? 1 : 0}
       style={{
         transform: transformStyle,
         transition: `transform ${delayBeforeNotRenderingNotebook}ms, opacity ${delayBeforeNotRenderingNotebook}ms`,
+        zIndex: 2,
+        overflowY: "hidden",
       }}
       onTransitionEnd={handleTransitionEnd}
     >
       {shouldShowNotebook && (
-        <Box className={NC.main}>
+        <Box
+          miw={{ lg: minNotebookWidth }}
+          style={{ flex: 1, overflowY: "auto" }}
+        >
           <Notebook {...props} />
         </Box>
       )}
 
-      {isNativePreviewSidebarOpen && (
-        <aside
-          className={NC.sqlSidebar}
-          data-testid="native-query-preview-sidebar"
+      {isNativePreviewSidebarOpen && windowWidth < 1280 && (
```

Comment:
It'd be nice to extract the 1280 as a constant. Also `minNotebookWidth`, `minSidebarWidth`, `handleWidth` and 1px border width which is mentioned in a comment (in `Handle`) but deserves to be a constant.